tests/sockets_cores_threads_test.go

```diff
@@ -0,0 +1,493 @@
+			Expect(vmi121Pod.Name).To(HavePrefix("virt-launcher-"+vmi121.Name), "Pod's name should contain name of the VM associated with it")
+			Expect(vmi122Pod.Name).To(HavePrefix("virt-launcher-"+vmi122.Name), "Pod's name should contain name of the VM associated with it")
+
+			By("get pod-s YAML files")
+			podName121 := vmi121Pod.Name
+			podName122 := vmi122Pod.Name
+			outPodYAML121, _, err := ktests.RunCommandWithNS(ktests.NamespaceTestDefault, "oc", "get", "pod", podName121, "-o", "yaml")
+			Expect(err).ToNot(HaveOccurred())
+			outPodYAML122, _, err := ktests.RunCommandWithNS(ktests.NamespaceTestDefault, "oc", "get", "pod", podName122, "-o", "yaml")
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking resources in the pod")
+			var CPUresourcesRequests, CPUresourcesLimits int
+			cpuExist, CPUresourcesRequests, CPUresourcesLimits := parseYAMLConfig(outPodYAML121)
+			Expect(cpuExist).To(BeTrue())
+			Expect(CPUresourcesRequests == 2).To(BeTrue())
+			Expect(CPUresourcesLimits == 0).To(BeTrue())
+			cpuExist, CPUresourcesRequests, CPUresourcesLimits = parseYAMLConfig(outPodYAML122)
+			Expect(cpuExist).To(BeTrue())
+			Expect(CPUresourcesRequests == 2).To(BeTrue())
+			Expect(CPUresourcesLimits == 2).To(BeTrue())
+
+			By("Get vmi121 XML")
+			vmiXml121, err := ktests.GetRunningVirtualMachineInstanceDomainXML(
+				virtClient,
+				vmi121,
+			)
+
+			By("Get vmi122 XML")
+			vmiXml122, err := ktests.GetRunningVirtualMachineInstanceDomainXML(
+				virtClient,
+				vmi122,
+			)
+
+			By("vmi unmarshal")
+			domStat121 := &api.DomainSpec{}
+			domStat122 := &api.DomainSpec{}
+			err = xml.Unmarshal([]byte(vmiXml121), domStat121)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = xml.Unmarshal([]byte(vmiXml122), domStat122)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("check XML")
+
+			By("check XML - vmi121")
+			Expect(uint(domStat121.VCPU.CPUs) == 2).To(BeTrue())
+			Expect(uint(domStat121.CPU.Topology.Sockets) == 2).To(BeTrue())
+			Expect(uint(domStat121.CPU.Topology.Cores) == 1).To(BeTrue())
+			Expect(uint(domStat121.CPU.Topology.Threads) == 1).To(BeTrue())
+
+			By("check XML - vmi122")
+			Expect(uint(domStat122.VCPU.CPUs) == 2).To(BeTrue())
+			Expect(uint(domStat122.CPU.Topology.Sockets) == 2).To(BeTrue())
+			Expect(uint(domStat122.CPU.Topology.Cores) == 1).To(BeTrue())
+			Expect(uint(domStat122.CPU.Topology.Threads) == 1).To(BeTrue())
+
+			By("Clean old pods")
+			var cleanPods []*corev1.Pod
+			clean_pods(virtClient, cleanPods)
+
+		})
+	})
+
+	Context("test cases 1.3,2.1,2.2 Check the validity of the XML file if user didn’t set the CPU topology at all", func() {
+		It("[test_id:1488] [test_id:1490] [test_id:1489] [test_id:1488] testcases 1.3,2.1,2.2 Check the validity of the XML file if user didn’t set the CPU topology at all", func() {
+
+			var wg sync.WaitGroup
+			vm_index := 0
+
+			for sockets := 0; sockets < 3; sockets++ {
+				for cores := 0; cores < 3; cores++ {
+					for threads := 0; threads < 3; threads++ {
```

Comment:
I guess, it's better to start from 1?, I see 0 is never used or am I wrong?

Reply:
No, it uses. If zero, then code uses other template